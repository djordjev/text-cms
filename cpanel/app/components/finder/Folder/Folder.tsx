import { Link } from "@remix-run/react";
import React from "react";

export interface FolderProps {
  name: string;
}

export const Folder: React.FC<FolderProps> = (props) => {
  const { name } = props;

  return (
    <Link
      className="u-flex u-flex-col u-items-center u-group"
      relative="path"
      to={name}
    >
      <img
        alt={name}
        height={60}
        loading="lazy"
        src="/images/folder.png"
        width={60}
      />
      <span className="group-hover:u-underline">{name}</span>
    </Link>
  );
};
