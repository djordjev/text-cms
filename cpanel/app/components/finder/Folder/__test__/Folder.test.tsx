import { createRemixStub } from '@remix-run/testing';
import { render, screen } from '@testing-library/react';

import { Folder, FolderProps } from '../Folder';

test('ContextMenu', () => {
  const props: FolderProps = { name: 'random name' };

  const createComponent = () => {
    const RemixStub = createRemixStub([
      { path: '/', Component: () => <Folder {...props} /> }
    ]);

    return <RemixStub />;
  };

  const { container } = render(createComponent());

  expect(screen.getByText(props.name)).toBeInTheDocument();

  expect(container.childNodes).toMatchSnapshot();
});
