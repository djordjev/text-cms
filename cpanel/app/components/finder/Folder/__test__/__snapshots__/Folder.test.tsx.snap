// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContextMenu 1`] = `
NodeList [
  <a
    class="u-flex u-flex-col u-items-center u-group u-w-1z u-max-w-1z"
    href="/random name"
  >
    <img
      alt="random name"
      height="60"
      loading="lazy"
      src="/images/folder.png"
      width="60"
    />
    <span
      class="group-hover:u-underline u-max-w-full u-text-ellipsis u-overflow-hidden u-text-nowrap"
    >
      random name
    </span>
  </a>,
]
`;
